hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/wal/TestAsyncLogRolling.java

```diff
@@ -51,6 +60,56 @@ public static void setUpBeforeClass() throws Exception {
     AbstractTestLogRolling.setUpBeforeClass();
   }
 
+  public static class SlowSyncLogWriter extends AsyncProtobufLogWriter {
+
+    public SlowSyncLogWriter(EventLoopGroup eventLoopGroup, Class<? extends Channel> channelClass) {
+      super(eventLoopGroup, channelClass);
+    }
+
+    @Override
+    public CompletableFuture<Long> sync(boolean forceSync) {
+      try {
+        Thread.sleep(syncLatencyMillis);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+      return super.sync(forceSync);
```

Comment:
For AsyncProtobufLogWriter, we should not block the sync thread, I think here, we should return a new CompletableFuture, and when the CompletableFuture we get by calling super.sync completes, we add syncLatencyMillis delay and then complete it?